lib/rucio/client/uploadclient.py

```diff
@@ -139,15 +139,32 @@ def upload(self, items, summary_file_path=None):
             file_did = {'scope': file['did_scope'], 'name': file['did_name']}
             dataset_did_str = file.get('dataset_did_str')
 
+            rse = file['rse']
+            rse_settings = self.rses[rse]
+            is_deterministic = rse_settings.get('deterministic', True)
+            if not is_deterministic and not pfn:
+                logger.warning('PFN has to be defined for NON-DETERMINISTIC site.')
```

Comment:
I would prefer to give an error here instead of a warning because the upload of this iteration will not be continued